### PR TITLE
DB-9369 Remove schema inference for Parquet tables

### PR DESCRIPTION
**Pull request by dgomezferro**
Make Decimal width computation consistent
Implement extract weekday in native spark

### CODE REVIEW

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryArithmeticOperatorNode.java [hunk 1]
```diff
@@ -152,7 +153,7 @@ public ValueNode bindExpression( FromList	fromList, SubqueryList subqueryList, L
 				int charMaxWidth = leftDTS.getMaximumWidth();
 				precision += (2 * charMaxWidth);								
 				scale += charMaxWidth;								
-				maxWidth = precision + 3;
+				maxWidth = DataTypeUtilities.computeMaxWidth(precision, scale);
```
[the remark this answers is not in the document]
Comment (dgomezferro) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryArithmeticOperatorNode.java]: Good catch, I missed that branch.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OperatorToString.java [hunk 2]
```diff
@@ -317,11 +317,25 @@ else if (operand instanceof ExtractOperatorNode) {
...
+                                parameter = "W";
+                                break;
+                            case "WEEKDAY":
+                                parameter = "u";
+                                break;
+                            case "WEEKDAYNAME":
+                                parameter = "EEEE";
+                                break;
+                        }
+                        return format("date_format(%s, \"\") ", opToString2(uop.getOperand(), vars), parameter);
```
[the remark this answers is not in the document]
Comment (dgomezferro) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OperatorToString.java]: This was badly broken sorry, I'm not sure why I didn't catch while manually testing on the service.

I've fixed it and added a test that uses parquet External tables to force this codepath.

Again, thanks for the catch!

**Comment by jyuanca** [comment]
@dgomezferro can you please resolve conflicts?

File: hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java [hunk 3]
```diff
@@ -61,6 +63,45 @@ public static void cleanoutDirectory() {
         SpliceUnitTest.clearDirectory( getExternalResourceDirectory() );
     }
 
+    @Test
+    public void testNativeSparkExtractFunction() throws Exception {
+        try {
+            File tablePath = File.createTempFile("native-spark-table", ".tmp");
```
Comment (martinrupp) [inline hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java:69]: i made some changes in https://github.com/splicemachine/spliceengine/pull/3486 to assure these temporary files are removed again. i'm waiting on this review, so if your change is in before mine i might have to adjust this as well. (this is more a note for myself to adjust this as well).
for now, **tablePath.deleteOnExit();** should be sufficient

**Pull request merged**